bin/strata-cli/src/signet.rs

```diff
@@ -163,25 +163,30 @@ pub async fn sync_wallet(
     Ok(())
 }
 
-async fn apply_update_stream(wallet: &mut Wallet, mut rx: UnboundedReceiver<WalletUpdate>) {
+async fn apply_update_stream(
+    wallet: &mut Wallet,
+    mut rx: UnboundedReceiver<WalletUpdate>,
+) -> Result<(), UpdateError> {
     while let Some(update) = rx.recv().await {
         match update {
-            WalletUpdate::SpkSync(update) => {
-                wallet.apply_update(update).expect("update to connect")
-            }
-            WalletUpdate::SpkScan(update) => {
-                wallet.apply_update(update).expect("update to connect")
-            }
+            WalletUpdate::SpkSync(update) => wallet
+                .apply_update(update)
+                .map_err(|e| UpdateError::from(Box::new(e) as BoxedErr))?,
+            WalletUpdate::SpkScan(update) => wallet
+                .apply_update(update)
+                .map_err(|e| UpdateError::from(Box::new(e) as BoxedErr))?,
```

Comment:
Here you could have another shim.

```rs
impl UpdateError {
  fn from_err<E: foobar + Sync + Send>(e: E) -> Self {
    Self::from(Box::new(e) as BoxedError)
  }
}
```

So that you could just call it as `.map_err(UpdateError::from_err)`.